### ISSUE TITLE
Way to use weighted loss in config

### ISSUE THREAD

**Issue by v-goncharenko** [catalyst-team/catalyst#1068]
I use configs to train my nets and now I've faced imbalanced classification problem. It could be solved by training classifier with weighted loss and `pytorch` losses support weighting out of the box.

But I couldn't use them because they accept only tensor input. If I provide a list of desired weights via config it get parsed to regular python list, so loss constructor fails.

To solve this I've written own wrapper class that constructs tensor and passes it to loss:

```python
class WrappedCrossEntropyLoss(nn.CrossEntropyLoss):
    '''Implements standard CrossEntropyLoss, where weight param can be list, not tensor'''

    def __init__(self, weights: Optional[List] = None):
        weights_tensor = torch.tensor(weights) if weights is not None else None
        super().__init__(weights_tensor)
```

Then regular config works fine:

```yaml
    criterion_params:
        criterion: WrappedCrossEntropyLoss
        weights: [0.2491, 1.0]
```

Is is intentional behaviour and could `catalyst` somehow cast such arguments automatically?

I guess it could be solved by enhancing config parsing algorithm. In case you are interested in solving this I could create some PR.

**Comment by Scitator** [comment]
Catalyst secret feature - you could define `get_from_params` class method to work with Catalyst Registry nicely. For example, https://github.com/catalyst-team/catalyst/blob/4f9fcd9c046b387378de6d295e7722b40361a020/catalyst/contrib/models/hydra.py#L143 .
In this way, all your params from the config would be passed to this method. All u need is to create the object instance by your own.

**Comment by v-goncharenko** [comment]
Weeeellll... that works for a custom class, am I right?

If yes - how is it supposed to work with regular `pytorch` classes like `nn.CrossEntropyLoss`?

I mean now for all built-in classes I have to write wrapper class (regardless which mechanism I use for passing params from config).

It would be great if one could use all built-in classes in config directly without writing boilerplated wrappers. Is it possible?

**Comment by Scitator** [comment]
Config-way of object creation is no the pure PyTorch way :)
So, you need to write your wrappers.

btw, speaking of the`weights` example, I think it should work naturally thanks to python dynamic cast list -> tensor during the `__init__` in the class. If no, it looks like we need to PR to the PyTorch :)

another opinion, that I see - add eval option to the Catalyst Config API, for example
```yml
   criterion_params:
        criterion: WrappedCrossEntropyLoss
        weights: "torch.tensor([0.2491, 1.0])"
```
or
```yml
   criterion_params:
        criterion: WrappedCrossEntropyLoss
        weights: "[0.2491, 1.0])":torch.tensor
```

Nevertheless, we need to design it properly first. If you would like to make such contribution - feel free to write me in slack ;)

**Comment by v-goncharenko** [comment]
AFAIK, PyTorch always accepted only tensor as an input for loss classes (any other type just fails), so I believe they won't change this (although I will search their issues for details). So PR to PyTorch doesn't look like a probable solution.

So for now answer to my initial question is
**No - one couldn't use `weights` param of `nn.CrossEntropyLoss` directly from config (without some wrapper class).**

Regarding designing config syntax - I'll contact you =)
